tools/bumper/parser.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+type component struct {
+	Url      string   `yaml:"url"`
+	Commit   string   `yaml:"commit"`
+	Metadata string   `yaml:"metadata"`
+}
+
+type componentsConfig struct {
+	Components map[string]component `yaml:"components"`
+}
+
+func parseComponentsYaml(componentsConfigPath string) (*componentsConfig, error) {
```

Comment:
Don't return a pointer here it's not optimizing anything

Reply:
I return a pointer mainly because I get a pointer from the allocation `new(componentsConfig)`.
plus it keeps it similar to it's opposite function `updateComponentsYaml` which also gets a pointer:
```
func updateComponentsYaml(componentsConfigPath string, config *componentsConfig) error {
```

Reply:
But why do we need a pointer ? why do we use `new` ?
you can do
```golang
config := componentsConfig{}
yaml.Unmarshal(componentsData, &config)
```